File: pset6/sentimental-mario-more/mario.py
# TODO
import cs50


def generate_half_pyramid(h):
    for i in range(h):  # outer loop to print out each line(0th level)
        for j in range(h - 1 - i):  # 1st lvl inner loop-1 to print spaces
            print(" ", end="")
        for k in range(i + 1):  # 1st lvl inner loop-2 to print 1st half
            print("#", end="")
        print("  ", end="")  # gap betwwen the two half-pyramids
        for k in range(i + 1):  # 1st lvl inner loop-3 to print 2nd half
            print("#", end="")
        print("\n", end="")


height = -1
while height < 1 or height > 8:
    height = cs50.get_int("Height: ")

generate_half_pyramid(height)

File: pset1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int x, i, s, c, d;
    printf("Program which creates 2 adjacent pyramids of desired height and width:--\n");

//This is to accept input between 1 and 8 only:
    do
    {
        x = get_int("What should the height be?(Type a number between 1 & 8):");
    }
    while (x < 1 || x > 8);



//Outer for-loop to print each row:
    for (i = 1; i < x + 1; i++)
    {

        //Alignment:
        for (s = 1; s < x - i + 1; s++)
        {
            printf(" ");
        }


        //Inner for-loop to print each column:
        for (c = 1; c < i + 1; c++)
        {
            printf("#");
        }
        printf("  ");
        for (d = 1; d < i + 1; d++)
        {
            printf("#");
        }
        printf("\n");

    }
}

